### PR TITLE
add MoonPay 'Buy LBC' button

### PR DESCRIPTION
**Pull request by akinwale**

### CODE REVIEW

File: app/src/main/java/io/lbry/browser/ui/wallet/WalletFragment.java [hunk 1]
```diff
@@ -57,6 +68,10 @@
 
 public class WalletFragment extends BaseFragment implements SdkStatusListener, WalletBalanceListener {
 
+    private static final String MOONPAY_KEY = "c2tfbGl2ZV9ueVJqVXNDbE5pcnVSdnlCMkJLWW5JcFA5VnA3dWU=";
+    private static final String MOONPAY_URL_FORMAT =
+            "https://buy.moonpay.io?apiKey=pk_live_xNFffrN5NWKy6fu0ggbV8VQIwRieRzy&colorCode=%%232F9176&currencyCode=LBC&showWalletAddressForm=true&walletAddress=%s&externalCustomerId=%s";
```
Comment (tiger5226) [inline app/src/main/java/io/lbry/browser/ui/wallet/WalletFragment.java:73]: Is it good to have these keys hardcoded like this? I would think these should be build environment variables.

Comment (tiger5226) [inline app/src/main/java/io/lbry/browser/ui/wallet/WalletFragment.java:73]: Is it good to have these keys hardcoded like this? I would think these should be build environment variables.

File: app/src/main/java/io/lbry/browser/ui/wallet/WalletFragment.java [hunk 2]
```diff
@@ -481,6 +505,48 @@ private boolean hasSkippedAccount() {
...
+        if (context != null) {
+            try {
+                SharedPreferences sp = PreferenceManager.getDefaultSharedPreferences(context);
+                receiveAddress = sp.getString(MainActivity.PREFERENCE_KEY_INTERNAL_WALLET_RECEIVE_ADDRESS, null);
+                if (Helper.isNullOrEmpty(receiveAddress)) {
+                    showError(getString(R.string.receive_address_not_set));
+                    return;
+                }
+
+                long userId = Lbryio.currentUser != null ? Lbryio.currentUser.getId() : 0;
```
Comment (tiger5226) [inline app/src/main/java/io/lbry/browser/ui/wallet/WalletFragment.java:520]: what happens if the user_id is 0 when passed to moonpay?

File: app/src/main/res/layout/card_wallet_balance.xml [hunk 3]
```diff
@@ -69,11 +69,22 @@
...
+            android:layout_marginLeft="12dp"
+            android:layout_marginRight="12dp"
+            android:textSize="16sp" />
+
         <TextView
             android:id="@+id/wallet_hint_convert_credits"
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
-            android:layout_marginTop="16dp"
+            android:layout_marginTop="6dp"
```
Comment (tiger5226) [inline app/src/main/res/layout/card_wallet_balance.xml:87]: intentional?

File: app/src/main/java/io/lbry/browser/ui/wallet/WalletFragment.java (hunk 2 above)
Comment (akinwale) [reply app/src/main/java/io/lbry/browser/ui/wallet/WalletFragment.java:520]: Nothing bad happens. The `externalCustomerId` field is freeform and it's just a way we can use tell that a purchase was done on Android.

File: app/src/main/java/io/lbry/browser/ui/wallet/WalletFragment.java (hunk 1 above)
Comment (akinwale) [reply app/src/main/java/io/lbry/browser/ui/wallet/WalletFragment.java:73]: We can put this in the `build.gradle`, but this is also checked into source control and adds extra steps to retrieving the required values.

File: app/src/main/res/layout/card_wallet_balance.xml (hunk 3 above)
Comment (akinwale) [reply app/src/main/res/layout/card_wallet_balance.xml:87]: Yes.

**Pull request merged**